### PR TITLE
feat: Add ability to select instances by CPU manufacturer

### LINKED ISSUE #3071: Have a way to select provision instances based on CPU type (AMD vs. Intel)
### Tell us about your request

Some of our workloads perform very poorly on AMD CPUs. We would love a way to exclude all AMD CPU instances from provisioning. Example:
```
- key: karpenter.k8s.aws/cpu-brand
      operator: NotIn
      values: [amd]
```

### Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?

Today, we have to do something like this and list all the instance families.
```
- key: karpenter.k8s.aws/instance-family
      operator: NotIn
      values: [c6a, m6a, r6a, c5a, m5a, r5a, c6ad, m6ad, r6ad, m5ad, r5ad, r5ad]
```

### Are you currently working around this issue?

Above.

### Additional Context

_No response_

### Attachments

_No response_

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### LINKED ISSUE #3529: Add a label for CPU Manufacturer/Name
### Tell us about your request

Similar to how you provide `karpenter.k8s.aws/instance-gpu-manufacturer` it would be great to have one for CPU manufacterer. We have a use case where workloads are using intel specific features and want to target `i` nodes vs `a` nodes. Right now the labels show they are just arch amd64 for both, which means we need to define specific instance types and update them as new generations come out.

### Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?

We have come workloads that only want to go on intel based instances. Right now we have our own metadata to add this type of information but would be great if it was "free".

### Are you currently working around this issue?

During startup we parse the output of `lscpu` to determine `amd` vs `intel`

### Additional Context

_No response_

### Attachments

_No response_

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by NetanelK**
Fixes #3071  
Fixes #3529  

**Description**
Added ability to select instance by its CPU manufacturer, by using the label `karpenter.k8s.aws/instance-cpu-manufacturer`

**How was this change tested?**
Ran unit tests, updated codegen to include manufacturer information

**Does this change impact docs?**
- [X] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jonathan-innis — APPROVED** [review]
This seems great to me! I think our E2E tests [will fail now because we aren't exercising all of the labels that we have](https://github.com/aws/karpenter-provider-aws/blob/main/test/suites/integration/scheduling_test.go#L65). Can you add a test case [here](https://github.com/aws/karpenter-provider-aws/blob/main/test/suites/integration/scheduling_test.go#L81) for the CPU manufacturer label now?

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Comment by NetanelK** [comment]
> This seems great to me! I think our E2E tests [will fail now because we aren't exercising all of the labels that we have](https://github.com/aws/karpenter-provider-aws/blob/main/test/suites/integration/scheduling_test.go#L65). Can you add a test case [here](https://github.com/aws/karpenter-provider-aws/blob/main/test/suites/integration/scheduling_test.go#L81) for the CPU manufacturer label now?

I've added the required label there, is there any place that needs it too?

**Comment by jonathan-innis** [comment]
> I've added the required label there, is there any place that needs it too

Nope! That looks good to me. I'll approve, we'll wait for approval from another maintainer, but I don't expect there to be a huge issue here. Nice work!

**Review by jonathan-innis — DISMISSED** [review]
LGTM 🚀

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

File: website/content/en/docs/concepts/scheduling.md
Comment (engedaam) [inline website/content/en/docs/concepts/scheduling.md:155]: Can we remove this from `docs` folder? We only want this change in `preview` folder as it is not released yet.

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by engedaam — APPROVED** [review]
LGTM 🚀

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Comment by ecerulm** [comment]
For reference, example values of `karpenter.k8s.aws/instance-cpu-manufacturer` are `aws`,`intel`, `aws` and you can find them in the preview documentation  https://karpenter.sh/preview/reference/instance-types/#r7i-family

**Pull request merged**
